rust/blockstore/src/arrow/block/delta/single_column_storage.rs

```diff
@@ -143,7 +149,60 @@ impl<V: ArrowWriteableValue<SizeTracker = SingleColumnSizeTracker>> SingleColumn
         }
     }
 
-    pub(super) fn split<K: ArrowWriteableKey>(
+    pub(super) fn split_before_size<K: ArrowWriteableKey>(
+        &self,
+        max_size: usize,
+    ) -> (CompositeKey, SingleColumnStorage<V>) {
+        let mut inner = self.inner.write();
+        let mut left_size = inner.size_tracker;
+
+        let mut split_key = None;
+
+        {
+            let mut iter = inner.storage.iter();
+            while let Some((key, value)) = iter.next_back() {
```

Comment:
iterates backwards for better performance given the assumption that the distribution of value sizes does not skew towards the front